src/OSPSuite.Presentation/Presenters/ParameterIdentifications/MultipleParameterIdentificationFeedbackPresentersManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿using OSPSuite.Core.Domain.ParameterIdentifications;
+using OSPSuite.Core.Events;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Container;
+using OSPSuite.Utility.Events;
+
+namespace OSPSuite.Presentation.Presenters.ParameterIdentifications
+{
+   public interface IMultipleParameterIdentificationFeedbackPresentersManager : 
+      IListener<ParameterIdentificationStartedEvent>,
+      IListener<ParameterIdentificationTerminatedEvent>,
+      IListener<ProjectClosedEvent>
+   {
+      IParameterIdentificationFeedbackPresenter FeedbackPresenterFor(ParameterIdentification parameterIdentification);
+   }
+
+   public class MultipleParameterIdentificationFeedbackPresentersManager : IMultipleParameterIdentificationFeedbackPresentersManager
+   {
+      static private IContainer _container;
+      private Cache<ParameterIdentification, IParameterIdentificationFeedbackPresenter> _currentParameterIdentifications = new Cache<ParameterIdentification, IParameterIdentificationFeedbackPresenter>();
+
+      public MultipleParameterIdentificationFeedbackPresentersManager(IContainer container)
+      {
+         _container = container;
+      }
+
+      public IParameterIdentificationFeedbackPresenter FeedbackPresenterFor(ParameterIdentification parameterIdentification)
+      {
+         if (_currentParameterIdentifications.Contains(parameterIdentification))
```

Comment:
I fear that this might be called from multiple threads no? Should it be thread safe?